org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterBatteryAlarm.java

```diff
@@ -154,17 +154,17 @@ public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
     }
 
     @Override
-    public void attributeUpdated(ZclAttribute attribute) {
+    public void attributeUpdated(ZclAttribute attribute, Object val) {
         if (attribute.getCluster() == ZclClusterType.POWER_CONFIGURATION
                 && attribute.getId() == ZclPowerConfigurationCluster.ATTR_BATTERYALARMSTATE) {
 
             logger.debug("{}: ZigBee attribute reports {}", endpoint.getIeeeAddress(), attribute);
 
-            // The value is a 32-bit bitmap, represented by an Integer
-            Integer value = (Integer) attribute.getLastValue();
-            if (value == null) {
+            if (!(val instanceof Integer)) {
```

Comment:
I see these ```instanceof``` checks throughout this code - do you think that's really necessary? It can only either be (in this case) Integer or null so I'm not sure we need to change all these null checks?